src/encoded/static/components/motifs.js

```diff
@@ -285,6 +351,59 @@ export const Motifs = (props) => {
         alignedEndCoordinate = Math.max(p.end, alignedEndCoordinate);
     });
 
+    // trimming the reference sequence to widest window of pwms
+    const referenceLength = alignedEndCoordinate - alignedStartCoordinate;
+    const referenceStart = alignedStartCoordinate - props.context.sequence.start;
+    const referenceSequence = props.context.sequence.sequence.slice(referenceStart, referenceStart + referenceLength);
+
+    const trackScrolling = () => {
+        if (tableTop && tableTop.current && refContainer && refContainer.current && refPlaceholder && refPlaceholder.current) {
+            if (tableTop.current.getBoundingClientRect().top <= 27) {
+                refContainer.current.classList.add('fixed-position');
+                refPlaceholder.current.setAttribute('style', `height:${refContainer.current.getBoundingClientRect().height}px`);
+            } else {
+                refContainer.current.classList.remove('fixed-position');
+                refPlaceholder.current.setAttribute('style', 'width:0px');
+            }
+        }
+    };
+
+    React.useEffect(() => {
+        if (limit === 0) {
+            document.addEventListener('scroll', trackScrolling);
+        }
+
+        // drawing reference sequence as a pwm so it lines up with the others
+        require.ensure(['d3', 'd3-sequence-logo'], (require) => {
+            setD3Lib(require('d3'));
+            const sequenceLogos = logos; // This is for local development when changes are needed to d3-sequence-logo.
+
+            const refSeq = referenceSequence.split('');
+            const fakePWM = [];
+
+            refSeq.forEach((nucleotide) => {
+                const maxHeight = 1000;
+                if (nucleotide === 'A') {
+                    fakePWM.push([maxHeight, 0, 0, 0]);
+                } else if (nucleotide === 'C') {
+                    fakePWM.push([0, maxHeight, 0, 0]);
+                } else if (nucleotide === 'G') {
+                    fakePWM.push([0, 0, maxHeight, 0]);
+                } else {
+                    fakePWM.push([0, 0, 0, maxHeight]);
+                }
+            });
+
+            if (d3lib) {
+                sequenceLogos.entryPoint(refToReference.current, fakePWM, d3lib, alignmentCoordinate, alignedStartCoordinate, referenceStart, '+', true);
+            }
+
+            return () => {
+                document.removeEventListener("scroll", trackScrolling);
+            }
+        });
+    }, [d3lib, props.limit, props.classList, props.context]);
```

Comment:
You’d normally use the variables you actually use in the `useEffect` callback, so since you assign `+props.limit` to limit, you can just use `limit` here. It doesn’t look like you use `classList` or `context` at all in the callback, so you don’t need those. You do need `refSequence`, `d3lib`, `refToReference`, `alignmentCoordinates`, `alignedStartCoordinate`, `referenceStart`. You might also need `trackScrolling` if ESLint complains. You also can’t use objects in the dependencies unless you want to check that you have a new copy of the object. The React dependency checks are very simple minded. It won’t detect that a value within an object has changed.